### PR TITLE
feat(html): allow referencing headers

### PR DESCRIPTION
**Pull request by dominique-pfister**
Allows using response headers in the indexed properties

### CODE REVIEW

File: src/html_json.js [hunk 1]
```diff
@@ -13,18 +13,24 @@ const moment = require('moment');
...
 const { IndexConfig } = require('@adobe/helix-shared');
-const fetchAPI = require('@adobe/helix-fetch');
-
-const { fetch } = process.env.HELIX_FETCH_FORCE_HTTP1
-  ? fetchAPI.context({ alpnProtocols: [fetchAPI.ALPN_HTTP1_1] })
+const { fetch } = require('@adobe/helix-fetch').context({
+  httpsProtocols:
   /* istanbul ignore next */
-  : fetchAPI;
+    process.env.HELIX_FETCH_FORCE_HTTP1 ? ['http1'] : ['http2', 'http1'],
```
Comment (tripodsan) [inline src/html_json.js:19]: ?? why ?

File: src/html_json.js [hunk 2]
```diff
@@ -13,18 +13,24 @@ const moment = require('moment');
...
+});
```
Comment (tripodsan) [inline src/html_json.js:20]: this project is already using helix-fetch v2:
https://github.com/adobe/helix-index-pipelines/blob/a884581c26cfd913e0f863334fb8d7ffa2076e16/package.json#L34

so keep the `alpnProtocols` way of  forcing http/1.1

Comment (dominique-pfister) [reply src/html_json.js:20]: Somehow, this doesn't work: in the nock statements in `html_json.test.js`:
```
  before(async () => {
    nock('https://main--helix-index-pipelines--adobe.project-helix.page')
      .get((uri) => {
-->     return uri.startsWith('/test/specs/hlx_up');
      })
      .reply(200, (uri) => {
        const path = p.resolve(SPEC_ROOT, 'hlx_up', p.basename(uri).replace(/\.md$/, '.html'));
        return fse.readFile(path, 'utf-8');
      }, { 'last-modified': 'Mon, 22 Feb 2021 15:28:00 GMT' })
      .persist();
  });
```
the line marked `-->` never gets executed, although the hostnames *do* match

Comment (dominique-pfister) [reply src/html_json.js]: OK, apparently passing `forceHttp1` as optional parameter to the service function **does** the trick, so I reverted the ALPN change, now tests are passing.

File: src/html_json.js [hunk 3]
```diff
@@ -200,60 +215,72 @@ function indexSingle(path, document, index, log) {
...
+  let fetchContext;
+  if (forceHttp1 || process.env.HELIX_PIPELINE_FORCE_HTTP1) {
+    fetchContext = fetchAPI.context({
+      alpnProtocols: [fetchAPI.ALPN_HTTP1_1],
+    });
+  } else {
+    /* istanbul ignore next */
+    fetchContext = fetchAPI;
+  }
+
```
Comment (dominique-pfister) [inline src/html_json.js:265]: as said, using ALPN setting again, but not as a direct require

File: src/html_json.js (hunk 2 above)
Comment (stefan-guggisberg) [reply src/html_json.js]: see 16486da

File: src/html_json.js [hunk 4]
```diff
@@ -94,29 +104,29 @@ async function fetchHTML(params, indices) {
...
-    if (!ret.ok) {
+    if (!resp.ok) {
       const message = body < 100 ? body : `${body.substr(0, 100)}...`;
-      log.warn(`Fetching ${url} failed: statusCode: ${ret.status}, message: '${message}'`);
-      return [url, { error: { reason: message, status: ret.status } }];
+      log.warn(`Fetching ${url} failed: statusCode: ${resp.status}, message: '${message}'`);
+      return [url, { error: { reason: message, status: resp.status } }];
     }
-    return [url, { response: body }];
+    return [url, { body, headers: resp.headers }];
```
Comment (stefan-guggisberg) [inline src/html_json.js:129]: Returning a copy of the headers would IMO be cleaner, i.e. 
`header: new Headers(resp.headers)`

**Review by stefan-guggisberg — APPROVED** [review]

File: src/html_json.js (hunk 4 above)
Comment (dominique-pfister) [reply src/html_json.js:129]: I see, where do I get a reference to the `Headers` class from? It is not defined in the `index.d.ts` file, only in `./fetch/headers.js`

Comment (stefan-guggisberg) [reply src/html_json.js:129]: `const { fetch, Headers } = require('@adobe/helix-fetch');`

**Review by tripodsan — APPROVED** [review]

**Pull request merged**